### PR TITLE
initial layout implementation - closes #2

### LINKED ISSUE #2: Include a text editing viewer.

### PR DESCRIPTION
**Pull request by phated**
I've added the application layout mechanisms and created some plugins that implement the Appbar, Sidebar and Editor for the application.  They currently only represent layout and structure, with a small amount of design.  The next step will be to hook up functionality.

ref #2, #4, #6

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+  {
+    register: require('./plugins/editor'),
+    options: {
+      initial: `function helloWorld(hello){\n  hello = 'world';\n}`
+    }
+  }
+];
+
+app.register(plugins, function(err){
+  console.log('registered', err, app);
```
Comment (monteslu) [inline client.js:29]: console logging :-1:

Comment (DarkmatterVale) [inline client.js:29]: Why is console logging bad? Wouldn't that be a good way to debug?

Comment (monteslu) [inline client.js:29]: just as a general rule its better to have a logging system in place that can be tuned for different levels of output.  This one is probably helpful this early in development :)

Comment (DarkmatterVale) [inline client.js:29]: Ah, I think I see what you are saying. Thanks for the explanation!

**Pull request merged**